igor_src/main.cpp

```diff
@@ -1602,7 +1605,15 @@ int main(int argc , char* argv[]){
 				Aligner v_aligner = Aligner(v_subst_matrix , v_gap_penalty , V_gene);
 				v_aligner.set_genomic_sequences(v_genomic);
 				try{
-					v_aligner.align_seqs(cl_path + "aligns/" +  batchname + v_align_filename , indexed_seqlist , v_align_thresh_value , v_best_only , v_left_offset_bound , v_right_offset_bound);
+					if (!gen_output_CDR3_data)
+						v_aligner.align_seqs(cl_path + "aligns/" +  batchname + v_align_filename , indexed_seqlist , v_align_thresh_value , v_best_only , v_left_offset_bound , v_right_offset_bound);
+					else{ //assume seqs are CDR3s
+						cout<<"Running on CDRsonly..."<<endl;
+						unordered_map<string,pair<int,int>> v_genomic_offset_bounds;
+						for(unordered_map<string,size_t>::const_iterator iter = v_CDR3_anchors.begin() ; iter != v_CDR3_anchors.end() ; iter++)
+							v_genomic_offset_bounds.emplace((*iter).first,make_pair( (*iter).second, (*iter).second ));
```

Comment:
I think there clearly is an error here, `(*iter).second` corresponds to the V gene anchor index, it is by definition positive (+ it is an unsigned integer). You're thus enforcing that V genes offsets are positive, which by definition is false if only the CDR3 is given...

Reply:
Ok my mistake on this one what you did here makes sense with what you've done in Aligner.cpp l.262 `min_offset=-genomic_offset_bounds.at((*iter).first).first` (if we forget about the -3 issue aforementioned).
However it makes the definition of genomic specific offsets inconsistent with the general offset definition. I'll would thus put this `-` in the high level main instead of here so everything remains consistent